encord/client.py

```diff
@@ -504,10 +506,32 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        for _ in range(10):
+            polling_response = self._querier.basic_getter(
+                DatasetDataLongPolling,
+                self._config.resource_id,
+                payload={
+                    "process_hash": process_hash,
+                    "ignore_errors": ignore_errors,
+                },
+            )
+
+            if polling_response["is_done"]:
+                return AddPrivateDataResponse.from_dict(polling_response["response"])
+            else:
+                time.sleep(1)
 
-        return AddPrivateDataResponse.from_dict(response)
+        raise Exception("Response not received")
```

Comment:
That is not descriptive enough and also we would throw some sort of `EncordException` (or whatever they are called) instead of throwing generic ones.

Reply:
See above: not sure why would we ever want to do this; we should loop for as long as the response is "please wait".

Reply:
Not relevant after switching to `while True`